pint/formatting.py

```diff
@@ -287,10 +291,64 @@ def format_compact(unit: UnitsContainer, registry: UnitRegistry, **options) -> s
         division_fmt="/",
         power_fmt="{}**{}",
         parentheses_fmt=r"({})",
+        registry=registry,
         **options,
     )
 
 
+dim_order = [
+    "[substance]",
+    "[mass]",
+    "[current]",
+    "[luminosity]",
+    "[length]",
+    "[time]",
+    "[temperature]",
+]
+
+
+def dim_sort(units: Iterable[list[str]], registry: UnitRegistry):
+    """Sort a list of units by dimensional order.
```

Comment:
might be good to mention the iso draft here?

Reply:
Good suggestion.  I've made mention of that spec in defaults_en.txt, but as far as the function is concerned, the dim order comes from the registry.  Which is also visible in the code.  I'll make that a bit more visible.